web-locks.js

```diff
@@ -11,21 +11,20 @@ const UNLOCKED = 1;
 let locks = null; // LockManager instance
 
 class Lock {
-  constructor({ name, mode = 'exclusive', buffer = null, timeout = null }) {
+  constructor(name, mode = 'exclusive', buffer = null) {
     this.name = name;
     this.mode = mode; // 'exclusive' or 'shared'
     this.queue = [];
     this.owner = false;
     this.trying = false;
-    this.timeout = timeout;
     this.buffer = buffer ? buffer : new SharedArrayBuffer(4);
     this.flag = new Int32Array(this.buffer, 0, 1);
     if (!buffer) Atomics.store(this.flag, 0, UNLOCKED);
   }
 
-  enter(handler) {
+  enter(handler, timeout) {
```

Comment:
That it much better, but we need to compare passing timeout and timer instance.

Reply:
> Maybe we need to use `clearTimeout` on abort?

Hmm, maybe it would be better to throw an error on timeout, similar to abort behavior? 
Also, move all resolve / reject invocations to one place.